### DISCUSSION TITLE
Host java-debug-adapter rebuild

### DISCUSSION THREAD

**Discussion by jvalkeal** [nvim-java/nvim-java#83] [General]
Could we host custom build of `java-debug-adapter` as it has same `vsix` issue than the `java-test`?

https://github.com/mason-org/mason-registry/blob/c3b5f9ed13d014bafa2d68efee7dc858fd8e639c/packages/java-debug-adapter/package.yaml#L13

**Comment by s1n7ax** [comment]
Ah. Didn't know they removed it. Thanks. Yep. This has to be done. I will create a new release job

**Reply by jvalkeal** [reply]
I was about to propose mason registry to host these custom builds but you already did it and it's a good idea to keep some of these java related things in a same place. I suppose MS finally wanted to block a loophole to download `vsix` from github as it's violation of TOS to download those from marketplace outside of vscode.

Both `microsoft/vscode-java-test` and `microsoft/vscode-java-debug` have these issues that release tags are not reproducible builds meaning that you can't trust in future that a release actually builds(this is due to their more or less broken eclipse p2 repo refs).

Thx a lot!

**Reply by s1n7ax** [reply]
Added it. I will update the mason registry soon but with a different name. vscode-java-debug follows different versiniong compared to java-debug jars. Mixing up those version will lead to a disaster so name will be different

https://github.com/nvim-java/vscode-java-debug-releases
